include/datadog/opentracing.h

```diff
@@ -29,7 +32,37 @@ struct TracerOptions {
   int64_t write_period_ms = 1000;
 };
 
+// Forward-declarations for Writer
+class SpanData;
+using Trace = std::unique_ptr<std::vector<std::unique_ptr<SpanData>>>;
+
+class HttpEncoder {
+ public:
+  HttpEncoder() {}
+  virtual ~HttpEncoder() {}
+
+  virtual std::string path() = 0;
+  virtual std::map<std::string, std::string> headers(std::deque<Trace> &traces) = 0;
```

Comment:
And add a comment that describes the shape of this. From looking at the impl it looks idempotent, which is good! If we add a comment saying so, maybe it'll stay that way!